machine/bootcmdline_test.go

```diff
@@ -24,5 +24,16 @@ var _ = Describe("BootCMDLine", func() {
 
 			Expect(string(b)).To(Equal("baz:\n    bar: \"\"\nconfig_url: foo bar\n"), string(b))
 		})
+		It("works if cmdline contains a dash or underscore", func() {
+			f, err := os.CreateTemp("", "test")
+			Expect(err).ToNot(HaveOccurred())
+			defer os.RemoveAll(f.Name())
+
+			err = os.WriteFile(f.Name(), []byte(`config-url="foo bar" ba_z.bar=""`), os.ModePerm)
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = DotToYAML(f.Name())
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
Astronger test would be to actually test that the file was parsed correctly. No errors could easily mean that we skipped the error.

Reply:
This is a good point, but out of scope of the PR methinks. This was about the parser erroring out when there was a dash in any of the cmdline entries, which broke the full parsing and just returned an error!